bundles/io/org.eclipse.smarthome.io.transport.serial/src/main/java/org/eclipse/smarthome/io/transport/serial/SerialPortEventListener.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.transport.serial;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Interface of a serial port event listener.
+ *
+ * @author Markus Rathgeb - Initial contribution
+ */
+@NonNullByDefault
+public interface SerialPortEventListener {
```

Comment:
Hm, you are right, this interface currently has only one function, so it could be a good candidate.
But personally I don't like to mark a listener as a function interface because it is about a special interface what a listener has to be provide and that could vary.

See also the [documentation of the functional interface annotation]:

> If a type is annotated with this annotation type, compilers are required to generate an error message unless:

and

> However, the compiler will treat any interface meeting the definition of a functional interface as a functional interface regardless of whether or not a FunctionalInterface annotation is present on the interface declaration.

The only real "win" is to get a compiler error if someone adds another function to that interface.
But if we ever want another function in that interface we will do it and have to remove that annotation again.
So, what is the big win to add it now?

[documentation of the functional interface annotation]: However, the compiler will treat any interface meeting the definition of a functional interface as a functional interface regardless of whether or not a FunctionalInterface annotation is present on the interface declaration.

Reply:
You are absolutely right. From the provider perspective it doesn't add real value but from the consumer's perspective, it does. The annotation benefits the provider because it will generate a compile error if the interface is not functional - and it benefits the consumer because it documents that the interface is meant to be used in lambdas. That's the only benefit. I am leaving it to you now.